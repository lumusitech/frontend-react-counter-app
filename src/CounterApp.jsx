import React, { useState } from 'react'
import PropTypes from 'prop-types'

const CounterApp = ({ value }) => {

  const [count, setCount] = useState(value);

  const handleAdd = () => {
    setCount(count + 1);
  }

  const handleSubtract = () => {
    setCount(count - 1);
  }

  const handleReset = () => {
    setCount(value);
  }

  return (
    <>
      <h1>Counter App</h1>
      <h2>{count}</h2>
      <button onClick={handleAdd}>+1</button>
      <button onClick={handleSubtract}>-1</button>
      <button aria-label="btn-reset" onClick={handleReset}>Reset</button>
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired
}
CounterApp.defaultTypes = {
  value: 0
}

export default CounterApp