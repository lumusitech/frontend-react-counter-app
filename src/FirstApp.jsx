import PropTypes from 'prop-types'

export const FirstApp = ({ title, subtitle, name }) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <p>{subtitle}</p>
      <p>{subtitle}</p>
      <p>{name}</p>
    </>

  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
  name: PropTypes.string
}

FirstApp.defaultProps = {
  title: "No hay title",
  subtitle: "No hay subtitle",
}
