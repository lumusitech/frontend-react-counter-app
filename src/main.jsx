import React from 'react'
import ReactDOM from "react-dom/client"
import CounterApp from './CounterApp'
import { FirstApp } from './FirstApp'

import './styles.css'

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    {/* <FirstApp title="Hola, soy Luciano" /> */}
    <CounterApp value={0} />
  </React.StrictMode>
) 