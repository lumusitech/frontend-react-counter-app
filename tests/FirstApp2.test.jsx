import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe("testing on FirstApp2 component", () => {

  const title = "Hola, soy Luciano"
  const subtitle = "Soy un subtitulo"

  test('should match with snapshot', () => {
    const { container } = render(<FirstApp name="Luciano" title={title} subtitle="" />)

    expect(container).toMatchSnapshot()
  })

  test('should shows message "Hola, soy Goku"', () => {
    render(<FirstApp title={title} />)
    expect(screen.getByText(title)).toBeTruthy()
    // Sirve para ver que se muestra en pantalla al momento
    // screen.debug()
  })

  test('should shows title in h1', () => {
    render(<FirstApp title={title} />)

    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(title)
  })

  test('should shows subtitle passed by props', () => {
    render(<FirstApp subtitle={subtitle} />)
    expect(screen.getAllByText(subtitle).length).toBe(2)
  })
})