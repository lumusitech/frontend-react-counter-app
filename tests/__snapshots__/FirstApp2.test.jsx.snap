// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing on FirstApp2 component should match with snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Luciano
  </h1>
  <p />
  <p />
  <p>
    Luciano
  </p>
</div>
`;
